src/schema/v2/infiniteDiscovery/discoverArtworks.ts

```diff
@@ -105,10 +105,18 @@ export const DiscoverArtworks: GraphQLFieldConfig<void, ResolverContext> = {
 
         result = await findSimilarArtworks(
           tasteProfileVector,
-          limit,
+          8,
+          excludeArtworkIds,
+          artworksLoader
+        )
+
+        // backfill with random curated picks if we don't have enough similar artworks
+        const randomArtworks = await getInitialArtworksSample(
```

Comment:
Maybe we can use `relatedArtworks` here? @egdbear

Reply:
Right, we're already using MLT with categorical similarity on OpenSearch (OS)... so I don't see why we would hit ES. The idea is to use a hybrid of visual and categorical similarity on the OS backend.